inc/saiswitch.h

```diff
@@ -2834,6 +2834,22 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_SWITCH_ISOLATE,
 
+    /**
+     * @brief Available IP Next hop group member entries
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_SWITCH_ATTR_AVAILABLE_IP_NEXT_HOP_GROUP_MEMBER_ENTRY,
```

Comment:
@JaiOCP , can we please address the comment from @rck-innovium  in previous PR https://github.com/opencomputeproject/SAI/pull/1763

pasting the below one here:

We need to understand the CRM types for "IP" nexthop group and nexthop group members. Today in SAI, nexthop group does not really differ based on the constituent nexthop's type.

Reply:
@JaiOCP , if this is corresponding to next-hop-group-member types, the types are SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY which is not differentiating IP or non-IP.
The IP, tunnel type etc. is for next-hop entry .

Reply:
> @JaiOCP , can we please address the comment from @rck-innovium in previous PR #1763
> 
> pasting the below one here:
> 
> We need to understand the CRM types for "IP" nexthop group and nexthop group members. Today in SAI, nexthop group does not really differ based on the constituent nexthop's type.

NHG differs based on the member types. This change was introduced using SAI_NEXT_HOP_GROUP_ATTR_HIERARCHICAL_NEXTHOP bool attribute. Default is IP+Tunnel for backward compatibility.


SAI_SWITCH_ATTR_AVAILABLE_IP_NEXT_HOP_GROUP_MEMBER_ENTRY is for the IP only NGH members.

Reply:
> @JaiOCP , if this is corresponding to next-hop-group-member types, the types are sai_next_hop_group_type_t which is not differentiating IP or non-IP. The IP, tunnel type etc. is for next-hop entry .

HI Rita,

We have the new attribute introduced by Hierarchical ECMP that depicts what kind of NHG it is. 
WIth HECMP we have NHG consists only IP+Tunnel NH member entries or IP only entries. Defaults are always IP+Tunnel hence the new attribute is added to get the IP only available member entries.

Reply:
then how to query for available NHG entries that for the type of "IP+Tunnel"?  And also total NHG entries?  Can we also clarify the usage of SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY vs this one?

Reply:
Available NHG is captured in the PR comments.

Available NHG entries would be queried using the existing attribute SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY. This is the existing attribute and is for IP+Tunnel NHG member entries.

New read only attribute  SAI_SWITCH_ATTR_AVAILABLE_IP_NEXT_HOP_GROUP_MEMBER_ENTRY will give the IP only NHG member entries.

Reply:
there can be a mix of "IP+Tunnel" plus "IP only" NHG entries existing, right?  in this case, can we define how the available "IP+tunnel" NHG entries can be queried?

Reply:
SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY, just to confirm, will it cover all sorts for NEXT_HOP member types? Sonic queries this currently and relies on this for creating any type of nexthop group member.

Reply:
Would it be possible to use the generic CRM for this as well? My thought:

- Add a read-only attribute to the nhg member which also has ``@isresource`` to reflect whether its group is L1 or L2.
- Use that attribute via the generic crm apis.
- This way it is possible to query for total, by not passing the attribute, or L1 or L2, by passing a value for the attribute.